### DISCUSSION TITLE
Lesson 4: Property byetcode not found on Contract class.

### DISCUSSION THREAD

**Discussion by karlMoreno** [smartcontractkit/full-blockchain-solidity-course-py#744] [Q&A] [answered]
With my code here:
import solcx
from solcx import compile_standard, install_solc
solcx.install_solc('0.6.0')
install_solc("0.6.0")

import json
from web3 import Web3

with open("./SimpleStorage.sol", "r") as file:
    simple_storage_file = file.read()

compiled_sol = compile_standard(
    {
        "language": "Solidity",
        "sources": {"SimpleStorage.sol": {"content": simple_storage_file}},
        "settings": {
            "outputSelection": {
                "*": {"*": ["abi", "metadata", "evm.bytecode", "evm.sourceMap"]}
            }
        },
    },
    solc_version="0.6.0",
)

with open("compiled_code.json", "w") as file:
    json.dump(compiled_sol, file)

#get bytecode
bytecode = compiled_sol["contracts"]["SimpleStorage.sol"]["SimpleStorage"]["evm"][
    "bytecode"
]["object"]

# get abi
abi = compiled_sol["contracts"]["SimpleStorage.sol"]["SimpleStorage"]["abi"]

#for connecting to ganache learn ganache command line
w3 = Web3(Web3.HTTPProvider("http://127.0.0.1:7545"))
chain_id = 5777
my_address = "0x579549B06c11491c1F487661713ABBc96017E9d5"
private_key = "0x923a7d7e6af39a689483f32c07dc578ce4f03054fd61dea00e34516afbd44b39" #add 0x to convert to hexadecimal

#Create the contract in python
SimpleStorage = w3.eth.contract(abi=abi, byetcode=bytecode)
print(SimpleStorage)

I am getting the error:
AttributeError: Property byetcode not found on Contract class. `Contract.factory` only accepts keyword arguments which are present on the Contract class

I don't understand what I'm getting wrong. I am using a docker to run everything on python 3.8 because cytoolz only works on that version so perhaps that's the problem?

**Comment by cromewar** [comment, marked as answer]
Hello @karlMoreno 

you have a typo on this line: 

`SimpleStorage = w3.eth.contract(abi=abi, byetcode=bytecode)`

the correct word is bytecode, no byetcode. Give it a try.

**Reply by karlMoreno** [reply]
Oh my goodness thanks so much! im such a fool.

**Reply by cromewar** [reply]
Could happen sometimes, not be so rude with yourself.
